state/protocol/badger/state.go

```diff
@@ -35,11 +36,12 @@ type State struct {
 	results storage.ExecutionResults
 	seals   storage.Seals
 	epoch   struct {
-		setups   storage.EpochSetups
-		commits  storage.EpochCommits
-		statuses storage.EpochStatuses
+		setups  storage.EpochSetups
+		commits storage.EpochCommits
 	}
-	versionBeacons storage.VersionBeacons
+	protocolState       storage.ProtocolState
+	protocolStateReader protocol.ProtocolState
```

Comment:
To me, this naming has much ambiguity. A few suggestions:

* frequently, we use the extension "DB" to indicate that something is  storage layer abstraction. So here, we would rename  `protocolState` -> `protocolStateDB`. I would go even one step further and use `protocolStateSnapshotsDB`.
* For `protocol.ProtocolState` I would use the variable name `protocolState`, which is consistent with most of the other usages


```suggestion
	protocolStateSnapshotsDB       storage.ProtocolState
	protocolState protocol.ProtocolState
```